src/conversion/model-to-view-converters.js

```diff
@@ -115,9 +117,13 @@ export function insertUIElement( elementCreator ) {
 			return;
 		}
 
-		const viewPosition = conversionApi.mapper.toViewPosition( data.range.start );
+		const mapper = conversionApi.mapper;
 
-		viewWriter.insert( viewPosition, viewElement );
+		viewWriter.insert( mapper.toViewPosition( data.range.start ), viewElement );
+
+		if ( !data.range.isCollapsed ) {
+			viewWriter.insert( mapper.toViewPosition( data.range.end ), viewElement.clone( true ) );
```

Comment:
We've been discussing that it would be nice if closing element is different than opening element, so this information would have to be send to creator and two view elements would have to be created.

If that is not too complicated, it would be nice to have this, but it is not a strict requirement. Take both model->view and view->model conversion into consideration.